libcontainer/rootfs_linux.go

```diff
@@ -247,6 +247,15 @@ func mountToRootfs(m *configs.Mount, rootfs, mountLabel string) error {
 		// bind mount won't change mount options, we need remount to make mount options effective.
 		// first check that we have non-default options required before attempting a remount
 		if m.Flags&^(unix.MS_REC|unix.MS_REMOUNT|unix.MS_BIND) != 0 {
+			// Make a copy of the previous mount flags, because in a user
+			// namespace we are not allowed to drop mount options from the host
+			// bindmount.
+			var statfs unix.Statfs_t
+			if err := unix.Statfs(dest, &statfs); err != nil {
```

Comment:
This still seems like the wrong thing to be doing for me -- it seems like someone higher up (i.e. the person/program generating the runc config) should be providing these options. If we do this here, it's probably worth masking off the non-protected bits (i.e. everything besides the *ATIME, NODEV, NOEXEC, and NOSUID bits).

Reply:
Ideally, yes. But there are a lot of users of runc, and quite a few of them have produced incorrect configurations. This is a case that not a lot of people seem to be aware of, so making it "just work" on this level seems like it's much more effective (though maybe some of the other maintainers don't agree).

But yes, we should mask off the non-protected bits.